cloud-functions/functions/src/thumbnailing.ts

```diff
@@ -0,0 +1,85 @@
+import {getStorage} from "firebase-admin/storage";
+import * as mupdf from "mupdf";
+import * as vips from "wasm-vips";
+import {getDocumentFromId} from "./storage.js";
+import {logger} from "firebase-functions";
+
+const PDF_CONVERSION_DPI = 250;
+const RATIO_WIDTH_HEIGHT = 210 / 297;
+
+/**
+ * Get the buffer of a document
+ * @param {string} documentId The id of the document
+ * @return {ArrayBuffer} The buffer of the document
+ */
+async function getDocumentBuffer(documentId: string): Promise<ArrayBuffer> {
+  const storage = getStorage();
+  const bucket = storage.bucket();
+  const originalFileRef = bucket.file(documentId);
+  const originalFile = (await originalFileRef.download())[0];
+  return originalFile.buffer;
+}
+
+/**
+ * Convert a pdf to a png image
+ * @param {ArrayBuffer} buffer The pdf to convert
+ * @return {Uint8Array} The png image
+ */
+function convertPdfToImage(buffer: ArrayBuffer): ArrayBuffer {
+  const originalDocument = mupdf.Document.openDocument(buffer, "application/pdf");
+  const page = originalDocument.loadPage(0);
+  const matrix = mupdf.Matrix.scale(PDF_CONVERSION_DPI / 72, PDF_CONVERSION_DPI / 72);
```

Comment:
What is this 72 ? It could be interesting to put it in a named constant

Reply:
Indeed, the 72 is the default DPI for PDFs in the context of converting it to an image. The other constant is the expected DPI for the output. I will rename them